.github/workflows/tests-clustermesh-upgrade.yaml

```diff
@@ -424,23 +428,79 @@ jobs:
             --conn-disrupt-dispatch-interval 0ms
 
 
-      - name: Upgrade Cilium in cluster1 and enable kvstoremesh
+      - name: Upgrade Cilium in cluster1
         env:
           KVSTORE_ID: 1
         run: |
           cilium --context ${{ env.contextName1 }} upgrade --reset-values \
             ${{ steps.newest-vars.outputs.cilium_install_defaults }} \
             ${{ steps.vars.outputs.cilium_install_defaults }} \
             ${{ steps.kvstore.outputs.cilium_install_kvstore }} \
-            ${{ steps.clustermesh-vars.outputs.cilium_install_cluster1 }} \
-            --set clustermesh.apiserver.kvstoremesh.enabled=true
+            ${{ steps.clustermesh-vars.outputs.cilium_install_cluster1 }}
 
-      - name: Rollout Cilium agents in cluster2
-        if: ${{ !matrix.external-kvstore }}
+      - name: Wait for cluster mesh status to be ready
         run: |
-          # This makes sure that the remote agents reconnect to the new instance of the
-          # clustermesh-apiserver, without waiting for the watchdog mechanism to kick in.
-          kubectl --context ${{ env.contextName2 }} rollout restart -n kube-system ds/cilium
+          cilium --context ${{ env.contextName1 }} status --wait
+          cilium --context ${{ env.contextName2 }} status --wait
+          cilium --context ${{ env.contextName1 }} clustermesh status --wait --wait-duration=5m
+          cilium --context ${{ env.contextName2 }} clustermesh status --wait --wait-duration=5m
+
+      - name: Write the Service manifest for testing failover
+        run: |
+          cat << EOF > echo-failover.yaml
+          apiVersion: v1
+          kind: Service
+          metadata:
+            annotations:
+              service.cilium.io/global: "true"
+            labels:
+              kind: echo
+              context: failover
+            name: echo-other-node-failover
+            namespace: cilium-test
+          spec:
+            ipFamilies:
+            - IPv4
+            - IPv6
+            ipFamilyPolicy: PreferDualStack
+            ports:
+            - name: http
+              port: 80
+              protocol: TCP
+              targetPort: 8080
+            selector:
+              name: echo-other-node
+            sessionAffinity: None
+            type: ClusterIP
+          EOF
+
+      - name: Restart clustermesh-apiserver and ensure client can connect to new Service
```

Comment:
It should not be performed when running in external kvstore mode, as meaningless in that case:

```yaml
if: ${{ !matrix.external-kvstore }}
```

The same applies to the three steps below.